planner/solver.go

```diff
@@ -489,20 +487,27 @@ func saveSolutions(ctx context.Context, c *iowrappers.RedisClient, req *Planning
 	return nil
 }
 
-// returns true if the new travel plan can be added to priority queue
-func isDuplicatedPlan(plans map[string]bool, newPlan PlanningSolution) bool {
-	jointPlaceIDs := jointPlaceIdsForPlan(newPlan)
-	if _, exists := plans[jointPlaceIDs]; !exists {
-		plans[jointPlaceIDs] = true
-		return true
+// isPlanDuplicate checks if ANY place in the plan is seen in previous results.
+func isPlanDuplicate(seenPlaces map[string]bool, newPlan PlanningSolution) bool {
+	for _, placeID := range newPlan.PlaceIDS {
+		if _, exists := seenPlaces[placeID]; exists {
+			return true
+		}
+	}
+	// mark all places in the plan as seen
+	for _, placeID := range newPlan.PlaceIDS {
+		seenPlaces[placeID] = true
 	}
 	return false
 }
 
-func jointPlaceIdsForPlan(newPlan PlanningSolution) string {
-	placeIDs := make([]string, len(newPlan.PlaceIDS))
-	copy(placeIDs, newPlan.PlaceIDS)
-	radix.Sort(placeIDs)
-	jointPlaceIDs := strings.Join(placeIDs, "_")
-	return jointPlaceIDs
+// removePlaces deletes all places of the input plan potentially stored in the input hashMap.
+func removePlaces(seenPlaces map[string]bool, plan PlanningSolution) {
+	for _, placeID := range plan.PlaceIDS {
+		if _, exists := seenPlaces[placeID]; !exists {
+			iowrappers.Logger.Warnf("Place %s does not exist in map seenPlaces", placeID)
```

Comment:
Nit: change the log level to `Debugf` and add the function name in the message.

Reply:
Done